src/build/__init__.py

```diff
@@ -233,102 +134,80 @@ def _parse_build_system_table(pyproject_toml: Mapping[str, Any]) -> dict[str, An
     return build_system_table
 
 
+def _wrap_subprocess_runner(runner: RunnerType, env: env.IsolatedEnv) -> RunnerType:
+    def _invoke_wrapped_runner(cmd: Sequence[str], cwd: str | None, extra_environ: Mapping[str, str] | None) -> None:
+        runner(cmd, cwd, {**(env.make_extra_environ() or {}), **(extra_environ or {})})
+
+    return _invoke_wrapped_runner
+
+
 class ProjectBuilder:
     """
     The PEP 517 consumer API.
     """
 
+    _DEFAULT_RUNNER = pyproject_hooks.default_subprocess_runner
```

Comment:
Is this class property really needed? A subclass can just inherit & overload `__init__`, changing the default of `runner`. This just simplifies this slightly at the expense of having two ways to do the same thing, with a private/harder to discover way being the added way.

Reply:
It was only because it is referenced both in `__init__` and `from_env` - the intention wasn't to simplify overwriting the runner (which you cannot do anyway in Python since function defaults are early-bound).  I could move it to the top level or double underscore it.

Reply:
Ahh, okay, that's fine then. I'd slightly prefer this be on the top level, as I could see someone wanting to use `class NewThing: _DEFAULT_RUNNER=other` to override this, which is what I'd usually see this pattern for. Does it really need to be aliased, couldn't it be left like before and just referenced twice?

Reply:
I've put it back.